cert/aziot-certd/src/lib.rs

```diff
@@ -235,10 +307,24 @@ impl UpdateConfig for Api {
             homedir_path: _,
             endpoints: _,
         } = new_config;
+
+        // Config change may have altered cert issuance. Reset the auto-renewed EST ID certs.
+        cert_renewal::engine::clear(&self.renewal_engine).await;
```

Comment:
```suggestion
        cert_renewal::engine::clear(self.renewal_engine.as_ref()).await;
```

Reply:
& should be fine here. Is there a reason to use as_ref instead?

Reply:
Yeah, autoderef is fine. No need to use `.as_ref()`